clusterloader2/pkg/chaos/nodes.go

```diff
@@ -98,11 +99,15 @@ func (k *NodeKiller) pickNodes() ([]v1.Node, error) {
 	rand.Shuffle(len(nodes), func(i, j int) {
 		nodes[i], nodes[j] = nodes[j], nodes[i]
 	})
-	numNodes := int(k.config.FailureRate * float64(len(nodes)))
-	if len(nodes) > numNodes {
-		return nodes[:numNodes], nil
+	expectedNodesToFail := k.config.FailureRate * float64(len(nodes))
+	nodesToFail := nodes[:0]
+	klog.Infof("%s: %d nodes available, expecting to fail %f nodes", k, len(nodes), expectedNodesToFail)
```

Comment:
+1 for adding more logs (I'd even add more, e.g. listing exactly which nodes were returned here)

Reply:
This information is available from 'kill' function in line 117, but I can duplicate that.